xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -1975,11 +1983,17 @@ where
             }
         });
 
-        self.handle_metadata_update_from_commit(payload.metadata_field_changes, storage)?;
+        self.handle_metadata_update_from_commit(&payload.metadata_field_changes, storage)?;
+
+        // When a DM is stitched, it can repeat group updates. We want to prevent saving those messages.
+        if self.update_already_exists(&payload, storage)? {
+            return Ok(None);
```

Comment:
Here is the main write check. Intentionally left the processing in place even if it's a duplicate, but maybe we want to put it after this short-circuit?